test/Worker.Extensions.DurableTask.Tests/FunctionsDurableTaskClientTests.cs

```diff
@@ -82,13 +92,159 @@ public void CreateHttpManagementPayload_WithHttpRequestData()
         }
 
+        /// <summary>
+        /// Test that the `WaitForCompletionOrCreateCheckStatusResponseAsync` method returns the expected response when the orchestration is completed.
+        /// The expected response should include OrchestrationMetadata in the body with an HttpStatusCode.OK.
+        /// </summary>
+        [Fact]
+        public async Task TestWaitForCompletionOrCreateCheckStatusResponseAsync_WhenCompleted()
+        {
+            string instanceId = "test-instance-id-completed";
+            var expectedResult = new OrchestrationMetadata("TestCompleted", instanceId)
+            {
+                CreatedAt = DateTime.UtcNow,
+                LastUpdatedAt = DateTime.UtcNow,
+                RuntimeStatus = OrchestrationRuntimeStatus.Completed,
+                SerializedCustomStatus = "TestCustomStatus",
+                SerializedInput = "TestInput",
+                SerializedOutput = "TestOutput"
+            };
+
+            var client = this.GetTestFunctionsDurableTaskClient( orchestrationMetadata: expectedResult);
+
+            HttpRequestData request = this.MockHttpRequestAndResponseData();
+
+            HttpResponseData response = await client.WaitForCompletionOrCreateCheckStatusResponseAsync(request, instanceId);
+
+            Assert.NotNull(response);
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+
+            // Reset stream position for reading
+            response.Body.Position = 0;
+            var orchestratorMetadata = await System.Text.Json.JsonSerializer.DeserializeAsync<dynamic>(response.Body);
+
+            // Assert the response content is not null and check the content is correct.
+            Assert.NotNull(orchestratorMetadata);
+            AssertOrhcestrationMetadata(expectedResult, orchestratorMetadata);
+        }
+
+        /// <summary>
+        /// Test that the `WaitForCompletionOrCreateCheckStatusResponseAsync` method returns expected response when the orchestrator didn't finish within
+        /// the timeout period. The response body should contain a HttpManagementPayload with HttpStatusCode.Accepted.
+        /// </summary>
+        [Fact]
+        public async Task TestWaitForCompletionOrCreateCheckStatusResponseAsync_WhenRunning()
+        {
+            string instanceId = "test-instance-id-running";
+            var expectedResult = new OrchestrationMetadata("TestRunning", instanceId)
+            {
+                CreatedAt = DateTime.UtcNow,
+                LastUpdatedAt = DateTime.UtcNow,
+                RuntimeStatus = OrchestrationRuntimeStatus.Running,
+            };
+
+            var client = this.GetTestFunctionsDurableTaskClient(orchestrationMetadata: expectedResult);
+
+            HttpRequestData request = this.MockHttpRequestAndResponseData();
+            CancellationTokenSource cts = new CancellationTokenSource(TimeSpan.FromSeconds(10));
```

Comment:
nit: we should add a `using` when creating `CancellationTokenSource` since it's a disposable object to avoid a memory leak. It shouldn't matter in this case since it's just test code, but it's a good habit to get into.

```suggestion
            using CancellationTokenSource cts = new CancellationTokenSource(TimeSpan.FromSeconds(10));
```